plugins/uniques.js

```diff
@@ -179,84 +234,125 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
 	}
 }
 
+window.plugin.uniques.setPortalAction = function(portal, action) {
+	var latE6 = portal.latE6;
+	var lngE6 = portal.lngE6;
+	var guid = window.findPortalGuidByPositionE6(latE6, lngE6);
+	var id = latE6 + "," + lngE6;
 
-window.plugin.uniques.setPortalVisited = function(guid) {
-	var uniqueInfo = plugin.uniques.uniques[guid];
-	if (uniqueInfo) {
-		if(uniqueInfo.visited) return;
+	if (guid) {
+		var uniqueInfo = window.plugin.uniques.uniques[guid];
+		if (!uniqueInfo) uniqueInfo = {};
+				// merge ALL pending actions, then remove from missedLatLngs
+		if (window.plugin.uniques.missedLatLngs[id]) {
+			Object.assign(uniqueInfo,window.uniques.missedLatLngs[id].action);
+			delete window.plugin.uniques.missedLatLngs[id];
+			window.plugin.uniques.storeLocal('missedLatLngs');
+		}
+		uniqueInfo[action] = true;
+		// special handling for captured
+		if (action === 'captured') uniqueInfo.visited = true;
```

Comment:
I'd prefer to see braces even in single-line conditions. Here and in numerous other places.